File: libs/modules/thermal/src/commonMain/kotlin/com/smellouk/konitor/thermal/ThermalInfo.kt
package com.smellouk.konitor.thermal

import com.smellouk.konitor.api.Info

data class ThermalInfo(
    val state: ThermalState,
    val isThrottling: Boolean,
    val temperatureC: Double = -1.0
) : Info {
    companion object {
        val INVALID = ThermalInfo(ThermalState.UNKNOWN, false)
        val UNSUPPORTED = ThermalInfo(ThermalState.UNSUPPORTED, false)
    }
}

File: libs/ui/flutter/lib/src/models/thermal_info.dart
class ThermalInfo {
  final String state;
  final bool isThrottling;
  final double temperatureC;

  const ThermalInfo({
    required this.state,
    required this.isThrottling,
    required this.temperatureC,
  });

  factory ThermalInfo.fromMap(Map<dynamic, dynamic> m) => ThermalInfo(
        state: m['state'] as String,
        isThrottling: m['isThrottling'] as bool,
        temperatureC: (m['temperatureC'] as num).toDouble(),
      );
}
